rllib/utils/test_utils.py

```diff
@@ -372,6 +347,88 @@ def check_compute_single_action(trainer,
                         "({})!".format(action, what, action_space))
 
 
+def check_learning_achieved(tune_results, min_reward, evaluation=False):
+    """Throws an error if `min_reward` is not reached within tune_results.
+
+    Checks the last iteration found in tune_results for its
+    "episode_reward_mean" value and compares it to `min_reward`.
+
+    Args:
+        tune_results: The tune.run returned results object.
+        min_reward (float): The min reward that must be reached.
+
+    Raises:
+        ValueError: If `min_reward` not reached.
+    """
+    # Get maximum reward of all trials
+    # (check if at least one trial achieved some learning)
+    avg_rewards = [(trial.last_result["episode_reward_mean"]
+                    if not evaluation else
+                    trial.last_result["evaluation"]["episode_reward_mean"])
+                   for trial in tune_results.trials]
+    best_avg_reward = max(avg_rewards)
+    if best_avg_reward < min_reward:
+        raise ValueError("`stop-reward` of {} not reached!".format(min_reward))
+    print("ok")
+
+
+def check_train_results(train_results):
+    """Checks proper structure of a Trainer.train() returned dict.
+
+    Args:
+        train_results: The train results dict to check.
+
+    Raises:
+        AssertionError: If `train_results` doesn't have the proper structure or
+            data in it.
+    """
+
+    from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+    from ray.rllib.utils.metrics.learner_info import LEARNER_INFO, \
+        LEARNER_STATS_KEY
+
+    # Assert that all the keys are where we would expect them.
+    for key in ["info", "hist_stats", "timers", "perf", "episode_reward_mean"]:
```

Comment:
should we define this list somewhere as a constant too?
also, should we make sure train_results doesn't contain any keys that we don't know here?
so next time someone adds a new key to the result set, this will make sure they update the known key list.

Reply:
Actually, this is just a pretty random list of keys that "should" be there, but it's not exhaustive and users can add any other keys to their custom Trainer's results dict.

For this test only, we could try adding more keys to this list to make sure our built-in algos are guaranteeing that at least these keys are always there.

Reply:
ok, I see.

Reply:
I added more keys, but again, user may add even more in their custom Trainer classes.
So this structure is not a very strict requirement. The focus in this test here is only on the "info" key, though.